src/framework/recycledList/RecycledList.jsx

```diff
@@ -0,0 +1,220 @@
+
+/**
+ * This component can be used to optimally-render the children of a list-type
+ * component (e.g. Grid, Menu).
+
+ * You just configure it with the root list component and the array of list
+ * items, and it will recycle a finite number of DOM elements to render the
+ * list items as the user scrolls.
+ */
+
+import React, {
+  Component,
+  PropTypes,
+} from 'react';
+
+export default class RecycledList extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      firstItemIndex: 0,
+      firstItemOffset: 0,
+      lastItemIndex: 0,
+      lastItemOffset: 0,
+    };
+
+    this.onScroll = this.onScroll.bind(this);
+  }
+
+  componentWillMount() {
+    this.updateItemPositions();
+  }
+
+  componentDidMount() {
+    this.props.scrollPosition.addListener(this.onScroll);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // When we receive more items, we need to update the recycled rows.
+    if (nextProps.items !== this.props.items) {
```

Comment:
Shouldn't the length be checked here - is this direct comparison actually working?